apps/dsaparam.c

```diff
@@ -179,6 +179,10 @@ int dsaparam_main(int argc, char **argv)
 
         data = app_malloc(len + 20, "BN space");
 
+        BIO_printf(bio_out, "#ifndef HEADER_DSA_H\n"
```

Comment:
The conditional guard around the header is redundant, the header includes this and there is no guarantee that the preprocessor symbol used isn't subject to change.

Just:
```
BIO_printf(bio_out, "#include <openssl/dsa.h>\n\n");
```

Reply:
So we remove the `#ifdef` from apps/dsaparam.c as well?

Reply:
This is `apps/dsaparam.c` isn't it?  If you meant `apps/dhparam.c`, then yes please.

Reply:
It's not unreasonable assumption that the output would be pasted into application code or #included. Hence it's not unreasonable to make application developer responsible for inclusion of corresponding header.

Reply:
Just in case it wasn't obvious, this is rather suggestion to *not* add #include to dsaparm.c and even omit #include from dhparam.c. The "paste" status is effectively implied by `static` qualifier.